include/ur_api.h

```diff
@@ -1949,9 +1978,10 @@ urMemGetNativeHandle(
 ///         + `NULL == phMem`
 UR_APIEXPORT ur_result_t UR_APICALL
 urMemCreateWithNativeHandle(
-    ur_native_handle_t hNativeMem, ///< [in] the native handle of the mem.
-    ur_context_handle_t hContext,  ///< [in] handle of the context object
-    ur_mem_handle_t *phMem         ///< [out] pointer to the handle of the mem object created.
+    ur_native_handle_t hNativeMem,                 ///< [in] the native handle to the memory.
+    ur_context_handle_t hContext,                  ///< [in] handle of the context object.
+    const ur_mem_native_properties_t *pProperties, ///< [in][optional] pointer to native memory creation properties.
```

Comment:
@kbenzie : since we dont have `urMemImageCreateWithNativeHandle`, then the only way for `urMemCreateWithNativeHandle` to know we are creating a memory or an image is by checking the pNext of properties, and checking for its sType == `UR_STRUCTURE_TYPE_MEM_IMAGE_NATIVE_PROPERTIES`, to confirm it is an image. This means that Properties is not optional when the native handle is an image.

Alternatively, we might need to define `urMemImageCreateWithNativeHandle` to specifically create images.

what do you think?

Reply:
So my thinking here was to reuse this entry point for both buffer and image native handle creation, on the buffer path the native image properties would be optional. Maybe that idea is flawed from a clarity of usage standpoint and a separate entry point should be the way forwards.

Reply:
If this were a separate entry point, I'd probably move the non-optional image format/desc to function arguments and reused the `ur_mem_native_properties_t` for `isNativeHandleOwned`.

Reply:
I've made this change, and renamed `urMemCreateWithNativeHandle` to `urMemBufferCreateWithNativeHandle` to match `urMemImageCreateWithNativeHandle`.